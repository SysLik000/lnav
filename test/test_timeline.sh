#! /bin/bash

export YES_COLOR=1

export TZ=UTC

touch -t 200711030923 ${srcdir}/logfile_glog.0
run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = 'test1' where log_line in (1, 3, 6)" \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = 'test1' where log_line in (1, 3, 6)" \
    -c ":hide-fields log_opid" \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = 'test1' where log_line in (1, 3, 6)" \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = '1234' where log_line in (1, 3, 6)" \
    -c ";UPDATE all_opids SET description = 'test-1' WHERE opid = '1234'" \
    -c ";UPDATE all_logs set log_opid = '5678' where log_line = 2" \
    -c ";UPDATE all_opids SET description = 'test-2' WHERE opid = '5678'" \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = '1234' where log_line in (1, 3, 6)" \
    -c ";UPDATE all_opids SET description = 'test-1' WHERE opid = '1234'" \
    -c ";UPDATE all_logs set log_opid = '5678' where log_line = 2" \
    -c ";UPDATE all_opids SET description = 'test-2' WHERE opid = '5678'" \
    -c ";SELECT * FROM all_opids" \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_generic.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':filter-in CdysLK1XpcrXOpVDuh' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':filter-out CdysLK1XpcrXOpVDuh' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':hide-file *' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':close *' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':hide-lines-before 2011-11-03 00:19:37' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':hide-lines-after 2011-11-03 00:20:30' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_strace_log.1
