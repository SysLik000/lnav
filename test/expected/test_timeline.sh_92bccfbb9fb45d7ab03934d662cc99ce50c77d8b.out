 2011-11-03T00:19        2m        2011-11-03T00:21 
                       26s250                    19
[1m[4m[35m   Duration   [0m[4m|[0m[4m [0m[1m[4m[31mâœ˜[0m[4m[33mâ–²[0m[4m [0m[4m|[0m[1m[4m[35m Operation[0m
[32m        23s045[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCN5[0m[7m[32mhnY3x51j6Hr1v4[0m[7m[32m  192.168.2.76[0m
[32m        32s389[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCmW[0m[7m[32mpC33jXuKpXNLcie[0m[7m[32m [0m[7m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCEh6Ka2HInkNSH0[0m[1m[7m[32m1[0m[1m[32mL2[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m         4s841[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCjPGiy13ncXKxU7[0m[1m[7m[32m65j[0m[1m[7m[32m 192[0m[1m[32m.168.2.76[0m
[32m         6s896[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCPoz7NUpXISemlN[0m[7m[32mSd[0m[7m[32m  192.1[0m[32m68.2.76[0m
[32m         6s838[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCt6ixh35y9AEr7J[0m[7m[32m7o9[0m[7m[32m 192.1[0m[32m68.2.76[0m
[1m[32m         6s890[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCaEFHq2HVQ5iGJQ[0m[1m[7m[32miD9[0m[1m[7m[32m 192.1[0m[1m[32m68.2.76[0m
[1m[32m         6s876[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCjinlH2fzDtvzI9[0m[1m[7m[32m637[0m[1m[7m[32m 192.1[0m[1m[32m68.2.76[0m
[32m         6s885[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCedw7H3ddE2yLiL[0m[7m[32moXc[0m[7m[32m 192.1[0m[32m68.2.76[0m
[32m         6s918[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCAUlC249svUfE6q[0m[7m[32m0g3[0m[7m[32m 192.1[0m[32m68.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCIJIDL1ULo4HpT24[0m[1m[7m[32mG[0m[1m[32ml[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m           017[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCLsqp41RLUd83arUQ[0m[1m[7m[32mb[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[32m           167[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCbCciH11995WKkobR[0m[7m[32m1[0m[32m [0m[32m192.168.2.76[0m
[32m           458[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mClcvKE1dqsEFQu46m[0m[7m[32m9[0m[7m[32m [0m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCaP2LpLGvsmX7yJO[0m[1m[32m  [0m[1m[7m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m           044[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCNbPns4mOMGgjI8Ele[0m[1m[7m[32m [0m[1m[32m192.168.2.76[0m
[32m           558[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mC185u7u9Q4qhJPhzl[0m[32m  [0m[7m[32m19[0m[32m2.168.2.76[0m
[32m           130[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCKzjfhsJ8vrn2rrfg[0m[32m  [0m[7m[32m1[0m[32m92.168.2.76[0m
[1m[32m           563[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCiIjAe1n5MnPOVpQ9f[0m[1m[32m 1[0m[1m[7m[32m9[0m[1m[32m2.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCGv2Tp4Ngt8MmKmVRd[0m[1m[32m 1[0m[1m[7m[32m9[0m[1m[32m2.168.2.76[0m
[32m         7s072[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mC5DisEMFU77Wk9Kae[0m[32m  1[0m[7m[32m92.168.2.7[0m[32m6[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCs5yEZ3ELZTeuTOsP4[0m[32m 192.[0m[7m[32m1[0m[32m68.2.76[0m
[1m[32m         4s668[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCu4gIx1BDNtGOl7Ht2[0m[1m[32m 192.[0m[1m[7m[32m168.2.7[0m[1m[32m6[0m
[1m[32m         1s289[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCRgW2I2zo3SInm6iT8[0m[1m[32m 192.16[0m[1m[7m[32m8.[0m[1m[32m2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCWJhMU2cTLEnseTmCb[0m[32m 192.16[0m[7m[32m8[0m[32m.2.76[0m
[32m           043[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCejI402rKGtdBXij4f[0m[32m 192.16[0m[7m[32m8[0m[32m.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mC2KnU34GcVV6amo8va[0m[1m[32m 192.16[0m[1m[7m[32m8[0m[1m[32m.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mC5vx4911iSMAJuShFd[0m[1m[32m 192.16[0m[1m[7m[32m8[0m[1m[32m.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCbUCgw1DrIGcXzONB7[0m[32m 192.16[0m[7m[32m8[0m[32m.2.76[0m
[32m           503[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mC96j2X1DixgLTj2Oi8[0m[32m 192.168[0m[7m[32m.2[0m[32m.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCYYyja3FFNEnftw3K6[0m[1m[32m 192.168.[0m[1m[7m[32m2[0m[1m[32m.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31mâ–ˆ[0m[1m[33m [0m[1m[32m  [0m[1m[32mCBHHuR1xFnm5C5CQBc[0m[1m[32m 192.168.[0m[1m[7m[32m2[0m[1m[32m.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCD1jfU3p9abEm77mzf[0m[32m 192.168.2.7[0m[7m[32m6[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mC0K9DaoPFkfnzwlZa[0m[32m  192.168.2.7[0m[7m[32m6[0m
[1m[32m           617[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCbQAWi3GX2bCmX5L56[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m           299[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCd8s2R3OGDgkhnvSu9[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[32m           206[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCBeaXe4Iyj1gXd2Iq[0m[32m  [0m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCmWpSw3VtjiAceBCwf[0m[32m [0m[32m192.168.2.76[0m
[1m[32m      1m12s202[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCbNCgO1MzloHRNeY4f[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m           316[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCX1GjC4vn52UY1uDv6[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCaPClb1Bf0RrRGtyWi[0m[32m [0m[32m192.168.2.76[0m
[32m        35s643[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCibfNy1QQW4ImDWRq5[0m[32m [0m[32m192.168.2.76[0m
[1m[32m      1m11s548[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCTRXSR3blXJE5ZE7Ij[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m      1m11s537[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCnGze54kQWWpKqrrZ4[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[32m        12s338[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mC3TZMB4CrUwYfkGJy1[0m[32m [0m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCK957ERTz8lBycly4[0m[32m  [0m[32m192.168.2.76[0m
[1m[32m         1s310[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCO5QKYQkcSdxQFA35[0m[1m[32m  [0m[1m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCurHpb1TGZOktTRNP1[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCuUKOQ1R3CqKBgeTdf[0m[32m [0m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mC3xkHgJnzZszVSTpi[0m[32m  [0m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCMrjgF2XLmRh9C9TR4[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mC2vQ8sVgyADHjtEda[0m[1m[32m  [0m[1m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCD69521bDXIAb4IkW[0m[32m  [0m[32m192.168.2.76[0m
[32m         6s649[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCC3vUI3gFB04zLvWRa[0m[32m [0m[32m192.168.2.76[0m
[1m[32m         2s667[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mC7Krri4g9tZfHniGXh[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCmxyBl2c8XAMTuHEk4[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[32m           500[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCSvs6v26bQqFylkk6l[0m[32m [0m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mC4pHul1H3OeWYz7o7i[0m[32m [0m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mC7Lcvr4vsTf6eYpBva[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCV8faD4L1sLL5kDwN9[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCxyAKs10ppnHFP6O8i[0m[32m [0m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mC6MrHk2C7rLuJqhjsg[0m[32m [0m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCvfUrT2DgYXXoZw9Ah[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCBX0254QJoklXNbvv2[0m[1m[32m [0m[1m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mC6Ym6jvMgikT0xTTc[0m[32m  [0m[32m192.168.2.76[0m
[32m              [0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32mCRNn9f1zKNlzHSM5pa[0m[32m [0m[32m192.168.2.76[0m
[1m[32m              [0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mCJLgi92kpp2gLgGTE5[0m[1m[32m [0m[1m[32m192.168.2.76[0m
